src/login/AzureLoginHelper.ts

```diff
@@ -100,14 +105,24 @@ export class AzureLoginHelper {
 		context.subscriptions.push(commands.registerCommand('azure-account.selectSubscriptions', () => this.selectSubscriptions().catch(console.error)));
 		context.subscriptions.push(this.api.onSessionsChanged(() => this.updateSubscriptions().catch(console.error)));
 		context.subscriptions.push(this.api.onSubscriptionsChanged(() => this.updateFilters()));
-		context.subscriptions.push(workspace.onDidChangeConfiguration(e => {
+		context.subscriptions.push(workspace.onDidChangeConfiguration(async e => {
 			if (e.affectsConfiguration(getSettingWithPrefix(cloudSetting)) || e.affectsConfiguration(getSettingWithPrefix(tenantSetting)) || e.affectsConfiguration(getSettingWithPrefix(customCloudArmUrlSetting))) {
 				const doLogin: boolean = this.doLogin;
 				this.doLogin = false;
 				this.initialize(e.affectsConfiguration(getSettingWithPrefix(cloudSetting)) ? 'cloudChange' : e.affectsConfiguration(getSettingWithPrefix(tenantSetting)) ? 'tenantChange' : 'customCloudARMUrlChange', doLogin)
 					.catch(console.error);
 			} else if (e.affectsConfiguration(getSettingWithPrefix(resourceFilterSetting))) {
 				this.updateFilters(true);
+			} else if (e.affectsConfiguration(getSettingWithPrefix(authLibrarySetting))) {
+				await this.logout();
```

Comment:
It's a bit jarring to me that it immediately logs out as soon as the setting is changed. Maybe we can make it a bit smoother? Like change the message to "You must sign out and reload the window..."? and only sign out if they click the button

The other option is to support both ADAL and MSAL within the same session. You would need to create both auth providers in the constructor, and then here you would just change which one is used. I feel like that would either "just work" or it would get super complicated and not be worth it.

Reply:
I went with option 1. Option 2 gets dicey because we'd still have to reload the window if they change the setting - otherwise they'd still be signed in using the old auth library when interacting with the extensions